regolith/helpers/a_proposalhelper.py

```diff
@@ -0,0 +1,190 @@
+"""Helper for adding a proposal to the proposals.yml collection.
+"""
+import datetime as dt
+import dateutil.parser as date_parser
+from dateutil.relativedelta import relativedelta
+import sys
+
+from regolith.helpers.basehelper import DbHelperBase
+from regolith.fsclient import _id_key
+from regolith.tools import (
+    all_docs_from_collection,
+    get_pi_id,
+)
+
+TARGET_COLL = "proposals"
+
+def subparser(subpi):
+    # Do not delete --database arg
+    subpi.add_argument("--database",
+                       help="The database that will be updated.  Defaults to "
+                            "first database in the regolithrc.json file."
+                       )
+    subpi.add_argument("name", help="A short but unique name for the proposal",
+                       )
+    subpi.add_argument("amount", help="value of award",
+                       )
+    subpi.add_argument("title", help="Actual title of the proposal"
+                       )
+    subpi.add_argument("--begin_date", help="The begin date for the proposed grant "
+                                          "in format YYYY-MM-DD.",
+                       default = 'tbd'
+                       )
+    subpi.add_argument("-d", "--duration", help="Duration of proposal in months",
+                       default = 'tbd'
+                       )
+    subpi.add_argument("--due_date", help="The due date for the proposal in format YYYY-MM-DD.",
+                       default = 'tbd'
+                       )
+    subpi.add_argument("-a", "--authors", nargs="+",
+                       help="Other investigator names", default = []
+                       )
+    subpi.add_argument("-c", "--currency",
+                       help="Currency in which amount is specified. Defaults to USD",
+                       default = 'USD'
+                       )
+    subpi.add_argument("-p", "--pi",
+                       help="ID of principal investigator. Defaults to"
+                            "group pi id in regolithrc.json"
+                       )
+    subpi.add_argument("--cppflag", help="Current and pending form (True or False). Defaults to True)",
+                       default = True
+                       )
+    subpi.add_argument("--other_agencies", help="Other agencies to which the proposal has been "
```

Comment:
I think this needs to be a string, so default should be ``"none"`` not ``False``

Reply:
The schema says it can be either a string or a boolean – would you prefer I edit this in the schema as well to only string?

Reply:
I think I wouldn't change the schema in this case.  If you give a default of ``"None"`` (the string) I think that is what we want to print into the cpp form though which seems easier (I think it should be capitalized.....sorry!)

Reply:
done!

Reply:
sorry, I may not have been so clear.  What I meant was that if none of the programs in regolith (e.g., builders etc.) uses "duration" we could remove it from the schema.  To check this using pycharm highlight ``duration`` and do CTL+SHIFT+F and it will show you everywhere in the project that the word is found (assuming you have opened regolith as a project).  This is quick if you have your pycharm set up right.

Then in the adder, I was thinking we could offer the user to add a begin date and a duration, or to add a begin date and an end date, but not both (or to avoid frustration, allow them to add both if they are consistent).  Make sure the help messages make this clear.

Does it make sense?